### PR TITLE
610/deposit token trade

### LINKED ISSUE #610: Implement deposit from trade widget
Implement deposit from trade widget

image

### PR DESCRIPTION
**Pull request by W3stside**
Waterfalls into #611 

Closes #610 

Deposit modal in TradeWidget for depositing all the things

Screenshot from 2020-06-22 13-09-41

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1139--dexreact.review.gnosisdev.com

File: src/components/TradeWidget/TokenRow.tsx [hunk 1]
```diff
@@ -170,6 +179,15 @@ const TokenRow: React.FC<Props> = ({
...
   })
 
+  const [visibleForm, showForm] = useState<'deposit' | void>()
+
+  // Checks innerWidth
+  const showResponsive = !!innerWidth && innerWidth < MEDIA.MOBILE_LARGE_PX
+  useNoScroll(!!visibleForm && showResponsive)
+
+  const isDepositFormVisible = visibleForm == 'deposit'
+  // const isWeth = balance.addressMainnet === WETH_ADDRESS_MAINNET
```
Comment (alfetopito) [inline src/components/TradeWidget/TokenRow.tsx:189]: ?

Comment (Velenir) [reply src/components/TradeWidget/TokenRow.tsx:189]: ??

**Comment by Velenir** [comment]
> I'd rather not display it when wallet is not connected.

Agree. Easier to just disable when not connected.

File: src/components/TradeWidget/TokenRow.tsx [hunk 2]
```diff
@@ -236,13 +254,31 @@ const TokenRow: React.FC<Props> = ({
...
           {!readOnly && (
-            // TODO: Implement deposit in Trade widget. When ready, show also the Wrap Ether button
-            //  https://github.com/gnosis/dex-react/issues/610
-            <Link className="btn" to="/wallet">
-              + Deposit
-            </Link>
+            <>
+              <div className="btn" onClick={(): void => showForm('deposit')}>
+                + Deposit
+              </div>
```
Comment (Velenir) [inline src/components/TradeWidget/TokenRow.tsx:262]: In `/wallet` page we allow depositing only when [token is enabled](https://github.com/gnosis/dex-react/blob/b477bb795a5864b10e0020b1b797df94746cc67b/src/components/DepositWidget/Row.tsx#L150).  Should do same here.

Plus can't deposit a not enabled token (without allowance)

File: src/components/TradeWidget/TokenRow.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/TradeWidget/TokenRow.tsx:189]: ?????????????

Comment (W3stside) [reply src/components/TradeWidget/TokenRow.tsx:189]: I think this is tired Leandro saying delete the fucking comment bro :joy:

Comment (alfetopito) [reply src/components/TradeWidget/TokenRow.tsx]: You are good at this :+1:

**Comment by alfetopito** [comment]
No more enable when disconnected \o/

But still able to deposit when token not enabled
screenshot_2020-06-23_09-12-43

Deposit should not be displayed until token is enabled.

**Comment by Velenir** [comment]
@alfetopito
Well yes
https://github.com/gnosis/dex-react/pull/1139#discussion_r444058027

File: src/components/TradeWidget/TokenRow.tsx (hunk 2 above)
Comment (alfetopito) [reply src/components/TradeWidget/TokenRow.tsx:263]: Unresolving

**Comment by anxolin** [comment]
image

I think the modal should close when it sends the transaction and provide a similar UX than in the balances. Otherwise you have the user blocked while the tx is being mined

File: src/components/TradeWidget/TokenRow.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/TradeWidget/TokenRow.tsx:265]: ok now @alfetopito @Velenir

Comment (alfetopito) [reply src/components/TradeWidget/TokenRow.tsx:268]: Niiiice!

Comment (Velenir) [reply src/components/TradeWidget/TokenRow.tsx:268]: Niiiiiiiiiiiice!

**Pull request merged**